pkg/api/v1beta3/types.go

```diff
@@ -934,7 +934,7 @@ type ServiceSpec struct {
 	Ports []ServicePort `json:"ports" description:"ports exposed by the service"`
 
 	// This service will route traffic to pods having labels matching this selector. If null, no endpoints will be automatically created. If empty, all pods will be selected.
-	Selector map[string]string `json:"selector" description:"label keys and values that must match in order to receive traffic for this service; if empty, all pods are selected, if not specified, endpoints must be manually specified"`
+	Selector string `json:"selector" description:"label keys and values that must match in order to receive traffic for this service; if empty, all pods are selected, if not specified, endpoints must be manually specified"`
```

Comment:
I don't follow the last bit here.  The idea of a magic label in the selector-map that was a full selector-expression is kind of smelly.  What's the downside of accepting either a map or a string?  A map is a logical AND of equality expressions.  If that is good enough for you, go ahead an use it.  The string (or some other type) is the escape valve for when that is not enough (and is the normalized form of all selectors).